format-avro/src/main/java/io/cdap/plugin/format/avro/input/AvroInputFormatProvider.java

```diff
@@ -117,32 +118,40 @@ public Schema getDefaultSchema(@Nullable FormatContext context) throws IOExcepti
     String filePath = conf.getProperties().getProperties().getOrDefault("path", null);
     SeekableInput seekableInput = null;
     FileReader<GenericRecord> dataFileReader = null;
+    GenericRecord firstRecord = null;
     try {
       Job job = JobUtils.createInstance();
       Configuration hconf = job.getConfiguration();
       // set entries here, before FileSystem is used
       for (Map.Entry<String, String> entry : conf.getFileSystemProperties().entrySet()) {
         hconf.set(entry.getKey(), entry.getValue());
       }
-      Path file = conf.getFilePathForSchemaGeneration(filePath, ".+\\.avro", hconf, job);
-      DatumReader<GenericRecord> dataReader = new GenericDatumReader<>();
-      seekableInput = new FsInput(file, hconf);
-      dataFileReader = DataFileReader.openReader(seekableInput, dataReader);
-      GenericRecord firstRecord;
-      if (!dataFileReader.hasNext()) {
-        return null;
+      List<Path> paths = conf.getFilePathsForSchemaGeneration(filePath, ".+\\.avro$", hconf, job); 
+      for (Path file : paths) {
+        DatumReader<GenericRecord> dataReader = new GenericDatumReader<>();
+        seekableInput = new FsInput(file, hconf);
+        try {
+          dataFileReader = DataFileReader.openReader(seekableInput, dataReader);
+          firstRecord = dataFileReader.next();
+          return new AvroToStructuredTransformer().convertSchema(firstRecord.getSchema());
+        } catch (Exception e) {
+          continue;
+        } finally {
+          if (dataFileReader != null) {
```

Comment:
Use `try` block to close resources instead of doing it in finally (https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)